tests/mock_network/mod.rs

```diff
@@ -219,64 +174,52 @@ fn multiple_joining_nodes() {
     });
-    let mut nodes = create_connected_nodes(&env, LOWERED_ELDER_SIZE);
 
-    while nodes.len() < 25 {
-        let initial_size = nodes.len();
-        info!("Size {}", nodes.len());
+    let iterations = 10;
+    let min_adds_per_iteration = 2;
+    let max_adds_per_iteration = 10;
 
-        let mut count_if_split_node = count_sections_members_if_split(&nodes);
-        let safe_section_size = unwrap!(nodes[0].inner.safe_section_size());
+    let mut rng = env.new_rng();
+    let mut nodes = create_connected_nodes(&env, LOWERED_ELDER_SIZE);
 
-        // Try adding five nodes at once, possibly to the same section. This makes sure one section
-        // can handle this, either by adding the nodes in sequence or by rejecting some.
-        // Ensure we do not create a situation when a recursive split will occur.
-        let count = 5;
-        for _ in 0..count {
-            if !can_accept_node_without_split(&count_if_split_node, safe_section_size) {
-                break;
-            }
+    for _ in 0..iterations {
+        let adds = rng.gen_range(min_adds_per_iteration, max_adds_per_iteration + 1);
+        let mut nodes_to_add: Vec<_> = (0..adds)
+            .map(|_| create_node_with_contact(&env, &mut nodes[0]))
+            .collect();
 
-            loop {
-                let node = create_node_with_contact(&env, &mut nodes[0]);
-                let valid_sub_prefix =
-                    new_node_prefix_without_split(&node, &count_if_split_node, safe_section_size);
-
-                if let Some(sub_prefix) = valid_sub_prefix {
-                    *unwrap!(count_if_split_node.get_mut(&sub_prefix)) += 1;
-                    nodes.push(node);
-                    break;
-                } else {
-                    info!("Invalid node {:?}, {:?}", node.name(), count_if_split_node);
-                }
-            }
-        }
-        let count = nodes.len() - initial_size;
+        info!(
+            "Simultaneously adding nodes: {}",
+            nodes_to_add.iter().map(|node| node.name()).format(", ")
+        );
 
+        nodes.append(&mut nodes_to_add);
         poll_and_resend(&mut nodes);
-        let removed_count = remove_nodes_which_failed_to_connect(&mut nodes, count);
-        let nodes_added: Vec<_> = nodes
-            .iter()
-            .rev()
-            .take(count - removed_count)
-            .map(TestNode::name)
-            .collect();
-        info!("Added Nodes: {:?}", nodes_added);
-        verify_invariant_for_all_nodes(&env, &mut nodes);
-        assert!(
-            !nodes_added.is_empty(),
-            "Should always handle at least one node"
-        );
+    }
+
+    // Verify all nodes joined.
+    for node in nodes.iter() {
+        assert!(node.inner.is_approved());
     }
 }
 
+#[test]
+fn single_split() {
+    let env = Environment::new(NetworkParams {
+        elder_size: LOWERED_ELDER_SIZE,
+        safe_section_size: LOWERED_ELDER_SIZE + 3,
```

Comment:
Why is it 7? (What is the context here in the test)

Reply:
So 7 is the smallest number where a certain bug was exposed  - the one caused by adults not handling the `MemberKnowledge` message. I think it's because it's the smallest number where when a split happens, the elders post-split might be completely different nodes than the elders pre-split which is exactly the situation which triggered the bug.

Reply:
OK. I think these things are good to have as comments in there when some magic number is used/updated. It becomes very obscure for someone looking at it otherwise.